src/utils/update/beneficial_owners_managing_officers_data_fetch.ts

```diff
@@ -34,24 +35,28 @@ const initialiseBoAndMoUpdateAppData = (appData: ApplicationData) => {
 };
 
 export const retrieveBeneficialOwners = async (req: Request, appData: ApplicationData) => {
+  const transactionId = appData.transaction_id!;
+  const overseasEntityId = appData.overseas_entity_id!;
   const pscs: CompanyPersonsWithSignificantControl = await getCompanyPsc(req, appData[EntityNumberKey] as string);
+  const boPrivateData = await getBeneficialOwnerPrivateData(req, transactionId, overseasEntityId);
```

Comment:
Please add some error handling, see https://github.com/companieshouse/overseas-entities-web/pull/1037/files#diff-4adb0c9659c385ca562375739dc7757cd4e725d13b3547a777e98330c8990a37R70

Reply:
Just an opinion here, is this error handling absolutely necessary. Error is handled already in the service call itself, it just adds extra nested levels

Reply:
It is also causing a code smell at the moment, complexity is too high due to all the if statements :(

Reply:
Yea, we can always extract methods out and put try/catch around them again

Reply:
@n-r-b fine to resolve ?